### PR TITLE
[AIRFLOW-1238] Decode URL-encoded characters.

### PR DESCRIPTION
**Pull request by matt2000**
URI username and password can contain URL-encoded characters.

Make sure you have checked _all_ steps below.

### JIRA
    - https://issues.apache.org/jira/browse/AIRFLOW-1283

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Forces username and password from URI to be decoded using urllib.parse module.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Trivial change to existing code.

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

Wow this is a tedious way to welcome a new contributor. How about I'll rebase the commit with the desired name schema once someone gives me an indication that it is likely to be accepted? :-)

### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

NA

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by Fokko** [comment]
@matt2000 The tests are failing.

**Comment by matt2000** [comment]
@Fokko I'm going to need some help then, because I don't see how my change could possibly because failures in, for example, `test_scheduler_sla_miss_email_exception` or `test_scheduler_sla_miss_callback_exception`. Maybe I'm not understanidng something? Can you point me in the right direction to troubleshoot?

**Comment by matt2000** [comment]
OK, I added a fallback for `None` and get good tests results, but now the minikube test fails although it didn't before now. I'm pretty sure this is failure of the minikube test environment, and not anything related to this code. Please advise on how to proceed?

File: airflow/models.py [hunk 1]
```diff
@@ -686,8 +686,8 @@ def parse_from_uri(self, uri):
         self.conn_type = conn_type
         self.host = hostname
         self.schema = temp_uri.path[1:]
-        self.login = temp_uri.username
-        self.password = temp_uri.password
+        self.login = unquote(temp_uri.username or '')
+        self.password = unquote(temp_uri.password or '')
```
Comment (ashb) [inline airflow/models.py:690]: Can you add an explicit test that passing a URI-quoted uri in works as expected please

**Review by ashb — CHANGES_REQUESTED** [review]

**Comment by ron819** [comment]
@matt2000 are you still working on this?

**Pull request closed without merging**